# Assignment 2_SET2

In [74]:
from scipy import stats
from scipy.stats import norm
import numpy as np

# Q.1


Ans:

In [7]:
#for normal distribution z = (X-μ)/б
z =(60-55)/8
z

0.625

In [26]:
#As we want to find the probability of service manager cannot meet his commitment,
#P(X>50) = 1-stats.norm.cdf(abs(z_score)) or 1-stats.norm.cdf(60, loc = mean, scale = std)
1-stats.norm.cdf(60, loc = 55, scale = 8)

0.26598552904870054

Inference: The probability that the service manager cannot meet his commitment is 0.26599

# Q.2


Ans:

In [66]:
#Z score for 38
lessthan_38 = stats.norm.cdf(38, loc = 38, scale = 6)
np.round(lessthan_38*100,2)

50.0

In [69]:
#Z score for 44
lessthan_44 = stats.norm.cdf(44, loc = 38, scale = 6)
np.round(lessthan_44*100,2)

84.13

In [70]:
#Hence People between 38 & 44  age
between38_44=stats.norm.cdf(44,38,6)-stats.norm.cdf(38,38,6)
np.round(between38_44*100,2)

34.13

Inference: statement A is False as Probability of employees aged between 38 to 44 is more.

In [72]:
# A training program for employees under the age of 30 at the center would be expected to attract about 36 employees.
under_30=stats.norm.cdf(30,38,6)
np.round(under_30*400,2)

36.48

Inference: statement B is True as no. of employees aged below 33 yrs attending training is 36

# Q.3

Ans:


If X1 ~ N(μ, σ2) and X2 ~ N(μ, σ2) are iid normal random variables,
then what is the difference between 2 X1 and X1 + X2

In [159]:
# The mean of  2X1 & X1 + X2 is same but the variance σ2 of 2X1 is 2 times more than variance of  X1 + X
# The difference between the two says that the two given variables are identically and independently distributed

# Q.4

Ans:

In [153]:
value_A_B= stats.norm.interval(0.99, loc=100, scale=20)
value_A_B

(48.48341392902199, 151.516586070978)

Inference: The two values of a and b,the probability of the random variable taking a value between them is 0.99: 48.5_151.5

# Q.5

Ans:

A) Specify a Rupee range (centered on the mean) such that it contains 95% probability for the annual profit of the company.

Profit1 ~ N(5, 3^2) and Profit2 ~ N(7, 4^2)

In [99]:
# Mean profits from two different divisions of a company = Mean1 + Mean2
Mean = 5+7
print('Mean Profit is',(Mean*45)/10, 'Crore Rupees')


Mean Profit is 54.0 Crore Rupees


In [100]:
#variance of two independent random variables.
SD = np.sqrt((9)+(16)) 
print('Standard Deviation is ',(SD*45)/10, 'Crore Rupees')

Standard Deviation is  22.5 Crore Rupees


In [101]:
#Rupee range,it contains 95% probability for the annual profit of the company.
print('Range is',(stats.norm.interval(0.95,54.0,22.5)),'crore Rupees')

Range is (9.900810347848783, 98.09918965215121) crore Rupees


B) Specify the 5th percentile of profit (in Rupees) for the company

In [102]:
#for 5th% of profit,we use the percent point function

In [123]:
percentile_5th = stats.norm.ppf(0.05, 54.0,22.5)
print(np.round(percentile_5th,),'crore Rupees')

17.0 crore Rupees


C) Which of the two divisions has a larger probability of making a loss in a given year?

In [129]:
# The probability of Division #1 making a loss
np.round((stats.norm.cdf(0,5,3))*100,2)

4.78

In [130]:
# The probability of Division #2 making a loss
np.round((stats.norm.cdf(0,7,4))*100,2)

4.01

Inference: Probability of Division 1 making a loss in a given year is more than Division 2.